tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -107,8 +107,34 @@ std::vector<std::vector<Long64_t>> GetFriendEntries(const std::vector<std::pair<
 
    return friendEntries;
 }
+
+////////////////////////////////////////////////////////////////////////
+/// Return the full path of the tree
+static std::string GetTreeFullPath(const TTree &tree)
+{
+   // Case 1: this is a TChain: we get the name out of the first TChainElement
```

Comment:
Why return the path of the first element rather than the path of the current element?

Reply:
because the processing did not start?

Reply:
Fair enough.  Might this routine be called at any other point in the processing? -> Answering my question: this is actually an internal routine, so no.

Reply:
:) Thanks philippe for these thoughts and commenting about them!!!!